src/shelf_usage/zone.cc

```diff
@@ -548,12 +550,22 @@ void Zone::free(Offset block) {
     // Do we really need to do fam_persist???
     //int64_t* b = (int64_t*) from_Offset(block);
     //fam_persist(b, find_size_from_level);
+    if (fast_alloc_ == true)
+      clear_data(block);
```

Comment:
With this change I see we can further optimize this function.. in clear_data we are creating zoneheader again, which may not be needed. Two options : 1. either to call find_size and memset_persist in Free itself (remove clear_data) or 2. Pass level and min_obj_size to clear_data.